dm/relay/local_reader.go

```diff
@@ -538,8 +550,11 @@ func (r *BinlogReader) parseFile(
 				state.latestPos = int64(e.Header.LogPos)
 				break
 			}
-			u, _ := uuid.FromBytes(ev.SID)
-			state.replaceWithHeartbeat, err = r.advanceCurrentGtidSet(fmt.Sprintf("%s:%d", u.String(), ev.GNO))
+			gtidStr, err2 := event.GetGTIDStr(e)
+			if err2 != nil {
+				return errors.Trace(err2)
```

Comment:
Seems we have abandoned trace

Reply:
I remember I have seen some zap logger printing error stack 🤔